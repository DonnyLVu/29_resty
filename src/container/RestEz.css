.RestEz {
  width: auto;
  background: rgb(235, 235, 235);
}