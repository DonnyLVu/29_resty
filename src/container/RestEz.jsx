import React, { Component } from 'react';
import Header from '../components/header/Header';
import Input from '../components/input/Input';
import Results from '../components/result/Results';
import { fetchApi } from '../services/Fetch';

export default class RestEz extends Component {
  state = {
    url: '',
    method: '',
    body: '',
    results: ''
  }

  handleChange = ({ target }) => {
    this.setState({ [target.name]: target.value });
  }

  handleSubmit = (e) => {
    e.preventDefault();
    this.fetch();
  }

  fetch = () => {
    const { url, method, body } = this.state;
    return fetchApi(url, method, body)
      .then(res => this.setState({ results: JSON.stringify(res) }));
  }

  render() {
    const { url, method, body, results } = this.state;

    return (
      <>
        <Header />
        <Input
          url={url}
          method={method}
          body={body}
          onSubmit={this.handleSubmit}
          onChange={this.handleChange} />
        <Results results={results} />
      </>
    );
  }
}

