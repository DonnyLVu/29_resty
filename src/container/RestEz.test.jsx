import React from 'react';
import { render } from '@testing-library/react';
import RestEz from './RestEz';

describe('RestEz component', () => {
  it('renders RestEz', () => {
    const { asFragment } = render(<RestEz />);
    expect(asFragment()).toMatchSnapshot();
  });
});
