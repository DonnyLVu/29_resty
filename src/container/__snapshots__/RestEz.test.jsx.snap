// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RestEz component renders RestEz 1`] = `
<DocumentFragment>
  <header
    class="Header"
  >
    <h1>
       RestEz
    </h1>
  </header>
  <section
    class="RestEz"
  >
    <form
      class="Input"
    >
      <input
        name="url"
        placeholder="This is the URL"
        type="text"
        value=""
      />
      <div>
        <label
          data-testid="get"
        >
          GET
        </label>
        <input
          name="method"
          type="radio"
          value="GET"
        />
        <label
          for="post"
        >
          POST
        </label>
        <input
          name="method"
          type="radio"
          value="POST"
        />
        <label
          for="put"
        >
          PUT
        </label>
        <input
          name="method"
          type="radio"
          value="PUT"
        />
        <label
          for="delete"
        >
          DELETE
        </label>
        <input
          name="method"
          type="radio"
          value="DELETE"
        />
      </div>
      <button>
        Hit it
      </button>
      <textarea
        name="body"
        placeholder="Raw JSON"
      />
    </form>
    <div
      class="Results"
    />
  </section>
</DocumentFragment>
`;
